src/main/java/org/isf/lab/gui/LabBrowser.java

```diff
@@ -473,14 +514,55 @@ class LabBrowsingModel extends DefaultTableModel {
 
 		private static final long serialVersionUID = 1L;
 
-		public LabBrowsingModel(String exam, LocalDate dateFrom, LocalDate dateTo) {
-			try {
-				pLabs = labManager.getLaboratory(exam, dateFrom.atStartOfDay(), dateTo.atStartOfDay());
-			} catch (OHServiceException e) {
-				pLabs = new ArrayList<>();
-				OHServiceExceptionUtil.showMessages(e);
-			}
-		}
+        public LabBrowsingModel(String exam, LocalDate dateFrom, LocalDate dateTo, String patid) {
+			boolean exists = true;
+            try {
+                Patient pat;
+                if (patid.isEmpty()) {
+                    pat = null;
+                } else {
+                    try {
+                        pat = patManager.getPatientById(Integer.parseInt(patid));
+						exists = pat == null ? false : true;
```

Comment:
```suggestion
						exists = pat != null;
```
This is much simpler and does the same thing.  The variable `exists` appears to be used only one in line 532.  Why not just test the variable `pat` there?    And if there is an exception (catch block at line 527) the variable `exists` will not be reset and will use the assigned value from line 518 which is `true`.   So, if there is an exception then line 533 will be executed which given that `pat` is `null` will most likely cause other errors.   It looks like some rethinking of this logic is in order.